libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1301,67 +1293,68 @@ void AP_InertialSensor::update(void)
 
             if (_startup_ms == 0) {
                 _startup_ms = AP_HAL::millis();
-            } else if (AP_HAL::millis()-_startup_ms > 2000) {
+            } 
+            else if (AP_HAL::millis() - _startup_ms > 2000) {
                 _startup_error_counts_set = true;
             }
         }
 
         for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            if (_accel_error_count[i] < _accel_startup_error_count[i]) {
+            if (_accel_error_count[i] <= _accel_startup_error_count[i]) {
                 _accel_startup_error_count[i] = _accel_error_count[i];
             }
-            if (_gyro_error_count[i] < _gyro_startup_error_count[i]) {
-                _gyro_startup_error_count[i] = _gyro_error_count[i];
-            }
-        }
-
-        // adjust health status if a sensor has a non-zero error count
-        // but another sensor doesn't.
-        bool have_zero_accel_error_count = false;
-        bool have_zero_gyro_error_count = false;
-        for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            if (_accel_healthy[i] && _accel_error_count[i] <= _accel_startup_error_count[i]) {
-                have_zero_accel_error_count = true;
```

Comment:
I need to re-think this, but pretty sure that someone tried to remove this before and it's wrong.
If I remind it correctly, this code is supposed to only set to the gyro/accel to unhealthy if there is another one with no errors; with your change you are always setting it to unhealthy no matter if there is another one or not.

Reply:
okay, I will revert this